src/pages/auth/Login.tsx

```diff
@@ -0,0 +1,126 @@
+import { useState } from 'react';
+import { useFormik } from 'formik';
+import MainContainer from '@/shared/MainContainer';
+import { ValidationSchemaLogin } from '@/shared/validation/schema/Login';
+import {
+  Box,
+  Button,
+  FormHelperText,
+  Grid,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { login } from '@/services/api/generated/endpoints';
+
+import { Form, Navigate } from 'react-router-dom';
+import { useMutation } from '@tanstack/react-query';
+import BackDropLoader from '@/components/common/BackDropLoader';
+import { setAuthorizationHeaders } from '@/services/api/mutator/axios-instance';
+import { authStore } from '@/store/authStore';
+
+const Login = () => {
+  const initialUserLoginData = {
+    email: '',
+    password: '',
+  };
+  const { setAccessToken } = authStore();
+  const [errorContainer, setErrorContainer] = useState<string | null>(null);
+
+  const { isLoading, mutate } = useMutation({
+    mutationFn: login,
+    onSuccess: (response) => {
+      setAccessToken(response.data.user.token);
+      setAuthorizationHeaders(response.data.user.token);
+    },
+    onError: (error) => {
+      const errorResponse = error as any;
+      const responseError =
+        errorResponse.response?.data?.errors?.['email or password']?.join('');
+      setErrorContainer(
+        `Email or Password ${responseError}` || 'An error occurred'
+      );
+      setTimeout(() => {
+        setErrorContainer(null);
+      }, 5000);
+    },
+  });
+
+  const formik = useFormik({
+    initialValues: initialUserLoginData,
+    validationSchema: ValidationSchemaLogin,
+    onSubmit: async (values) => {
+      let user = {
+        email: values.email,
+        password: values.password,
+      };
+      mutate({ user });
+      <Navigate to="/dashboard" />;
```

Comment:
@BardhMustafa I would suggest to use hooks insead of JSX element in this case of function please try const navigate = useNavigate(); and then navigate('/');